io-engine/src/rebuild/rebuild_job.rs

```diff
@@ -164,9 +165,19 @@ impl RebuildJob {
 
     /// Schedules the job to start in a future and returns a complete channel
     /// which can be waited on.
-    pub fn start(
+    pub async fn start(
         &self,
+        map: Option<RebuildMap>,
     ) -> Result<oneshot::Receiver<RebuildState>, RebuildError> {
+        let (s, r) = oneshot::channel();
```

Comment:
if map is None do we need to send at all?

Reply:
Fixed